### PR TITLE
Removed usage of half types in CTS gl tests

### LINKED ISSUE #1982: Eliminate usage of half type in the CTS test
Remove usage of half types in favor of float types and write_imagef in the kernels. Refer discussion on  CTS PR #1974

### PR DESCRIPTION
**Pull request by kamil-goras-mobica**
see #1982

### CODE REVIEW

File: test_conformance/gl/test_images_write_common.cpp [hunk 1]
```diff
@@ -37,8 +37,7 @@ static const char *kernelpattern_image_write_1D =
 "}\n";
 
 static const char *kernelpattern_image_write_1D_half =
-"#pragma OPENCL EXTENSION cl_khr_fp16 : enable\n"
-"__kernel void sample_test( __global half4 *source, write_only image1d_t dest )\n"
+"__kernel void sample_test( __global half *source, write_only image1d_t dest )\n"
 "{\n"
 "    uint index = get_global_id(0);\n"
 "    write_imagef( dest, index, vload_half4(index, (__global half *)source));\n"
```
Comment (bashbaug) [inline test_conformance/gl/test_images_write_common.cpp:43]: I think we can get rid of all of these casts to `__global half*` now.  For example:

```suggestion
"    write_imagef( dest, index, vload_half4(index, source));\n"
```

**Review by bashbaug — APPROVED** [review]
This is a great first step and we should merge this PR as-is.

I don't think we should close #1982 until we've looked at other uses of `supportsHalf` in these tests, though, since I think they have a similar problem.

**Comment by kamil-goras-mobica** [comment]
> This is a great first step and we should merge this PR as-is.
> 
> I don't think we should close #1982 until we've looked at other uses of `supportsHalf` in these tests, though, since I think they have a similar problem.

I saw in other files in gl directory that function supportsHalf() is used which is checking for cl_khr_fp16. I can take a look at them and create separate PR.

File: test_conformance/gl/test_images_write_common.cpp (hunk 1 above)
Comment (kamil-goras-mobica) [reply test_conformance/gl/test_images_write_common.cpp:43]: > I think we can get rid of all of these casts to `__global half*` now. For example:

@bashbaug Of course I will remove such necessary casting.

**Review by bashbaug — APPROVED** [review]

**Review by lakshmih — APPROVED** [review]

**Comment by bashbaug** [comment]
I've edited the PR comment and I don't think it will automatically close #1982 anymore.

**Pull request merged**